.github/workflows/heartbeat-tests.yml

```diff
@@ -0,0 +1,147 @@
+name: heartbeat-tests
+on:
+  workflow_dispatch: # As per documentation, the colon is necessary even though no config is required.
+  schedule:
+    # Cron syntax is "minute[0-59] hour[0-23] date[1-31] month[1-12] day[0-6]". '*' is 'any value', and multiple values
+    # can be specified with comma-separated lists. All times are UTC.
+    # So this expression means "run at 45 minutes past 1, 5, and 9 AM/PM UTC". The hours were chosen so that
+    # the jobs run only close to business hours of Central Time.
+    # Days were chosen to run only from Monday through Friday.
+    - cron: "45 13,17,21 * * 1,2,3,4,5"
+
+jobs:
+  production-heartbeat:
+    strategy:
+      # By default, if any job in a matrix fails, all other jobs are immediately cancelled. This makes the jobs run to completion instead.
+      fail-fast: false
+      matrix:
+        os: [{vm: ubuntu-latest, exe: .sh}, {vm: windows-2019, exe: .cmd}]
+        node: ['lts/*']
+    runs-on: ${{ matrix.os.vm }}
+    timeout-minutes: 60
+    steps:
+      # === Setup. We need to get the code, set up nodejs, java, and python, and create the results directory. ===
+      - uses: actions/checkout@v4
+        with:
+          ref: 'dev'
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node }}
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+      - uses: actions/setup-python@v5
+        with:
+        python-version: '3.10'
```

Comment:
I set the python version to match what our docs [require](https://developer.salesforce.com/docs/platform/salesforce-code-analyzer/guide/analyze-vscode.html#install-and-configure-on-your-local-computer).